src/modules/vendor.js

```diff
@@ -34,15 +35,30 @@ class Vendor {
         // Step 2: Submit signature
         if (!vendorData.vSignature) {
             const vmPubKeyMultihash = vendorData.vmPubKeyMultihash
-            this.chluIpfs.logger.debug(`Sending Vendor Signature for ${vmPubKeyMultihash}`)
+            this.chluIpfs.logger.debug(`Sending Vendor Signature for ${vmPubKeyMultihash}, profile: ${JSON.stringify(profile)}`)
             const signature = await this.chluIpfs.didIpfsHelper.signMultihash(vmPubKeyMultihash);
-            await this.sendSignature(url, didId, signature)
+            await this.sendSignature(url, didId, signature, profile)
         } else {
             this.chluIpfs.logger.debug('Marketplace/Vendor key already signed')
         }
+        // Step 3: (Optional) Submit Profile
+        if(profile) await this.updateProfile(url, profile)
         this.chluIpfs.logger.debug('Vendor Signup complete')
     }
 
+    async updateProfile(url, profile = null) {
+        const didId = this.chluIpfs.didIpfsHelper.didId
+        this.chluIpfs.logger.debug(`Updating Vendor Profile for Marketplace ${url} using DID ${didId}`)
+        const multihash = getDAGNodeMultihash(await createDAGNode(Buffer.from(JSON.stringify(profile))))
+        const signature = await this.chluIpfs.didIpfsHelper.signMultihash(multihash);
+        await this.sendProfile(url, didId, profile, signature)
+        this.chluIpfs.logger.debug(`Updating Vendor Profile for Marketplace ${url} using DID ${didId} updated`)
+    }
+
+    async sendProfile(url, didId, profile, signature) {
+        return await httpPOST(`${url}/vendors/${didId}/profile`, { profile, signature })
```

Comment:
What happens if the service on the other end is down? We should think about using a queue to decouple services. We could use Redis or just depend on SQS, so we don't have to maintain the queue daemon.

Thoughts?

Reply:
Good question. The service on the other end is a marketplace which creates all kinds of other problems if it goes down, I think we should have a discussion about it in general and how we can mitigate the damage if it happens